crates/stages/src/stages/hashing_storage.rs

```diff
@@ -91,62 +91,60 @@ impl<DB: Database> Stage<DB> for StorageHashingStage {
 
             // Aggregate all transition changesets and and make list of storages that have been
             // changed.
-            tx.cursor_read::<tables::StorageChangeSet>()?
-                .walk((from_transition, H160::zero()).into())?
-                .take_while(|res| {
-                    res.as_ref().map(|(k, _)| k.0 .0 < to_transition).unwrap_or_default()
-                })
-                .collect::<Result<Vec<_>, _>>()?
-                .into_iter()
-                // fold all storages and save its old state so we can remove it from HashedStorage
-                // it is needed as it is dup table.
-                .fold(
-                    BTreeMap::new(),
-                    |mut accounts: BTreeMap<Address, BTreeMap<H256, U256>>,
-                     (TransitionIdAddress((_, address)), storage_entry)| {
-                        accounts
-                            .entry(address)
-                            .or_default()
-                            .insert(storage_entry.key, storage_entry.value);
-                        accounts
+            tx.get_range::<tables::StorageChangeSet>(
```

Comment:
I would decouple this into smaller code bites to make it more readable

Reply:
I tried to resolve it w/ 89e761819bf4061735a80b13ea26048c4d265d9d. I just decouple the starting part because I'm not familiar with whole function chains. Is it acceptable?

Reply:
Ah, I see, the long chain existed prior to this PR too, I thought it was added because the diff lit all of it up. No worries! We should refactor it in a different PR though, it is hard to parse :smile: